crt_portal/cts_forms/views.py

```diff
@@ -178,6 +178,74 @@ def post(self, request, id):
         return render(self.request, 'forms/complaint_view/show/index.html', output)
 
 
+class ProFormView(SessionWizardView):
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+    def get_template_names(self):
+        return 'forms/pro_template.html'
+
+    def get_context_data(self, form, **kwargs):
+        context = super().get_context_data(form=form, **kwargs)
+
+        field_errors = list(map(lambda field: field.errors, context['form']))
+        page_errors = [error for field in field_errors for error in field]
+
+        # internal use only
+        ordered_step_names = [
+            'Intake',
+            'Contact',
+            'Service member',
+            'Primary concern',
+            'Follow up',
+            'Location',
+            'Personal characteristics',
+            'Date',
+            'Summary',
+        ]
+
+        context.update({
+            'field_errors': field_errors,
+            'page_errors': page_errors,
+            'num_page_errors': len(list(page_errors)),
+            'page_errors_desc': ','.join([f'"{error_desc}"' for error_desc in page_errors]),
+            'word_count_text': {
+                'wordRemainingText': _('word remaining'),
+                'wordsRemainingText': _(' words remaining'),
+                'wordLimitReachedText': _(' word limit reached'),
+            },
+            'ordered_step_names': ordered_step_names,
+            'stage_link': True,
+        })
+
+        return context
+
+    def done(self, form_list, form_dict, **kwargs):
+        form_data_dict = self.get_all_cleaned_data()
+        m2m_protected_class = form_data_dict.pop('protected_class')
+        m2m_hatecrime = form_data_dict.pop('hatecrimes_trafficking')
+        r = Report.objects.create(**form_data_dict)
+
+        # add a save feature for hatecrimes and trafficking question on primary reason page
+        # Many to many fields need to be added or updated to the main model, with a related manager such as add() or update()
+        for protected in m2m_protected_class:
+            p = ProtectedClass.objects.get(protected_class=protected)
+            r.protected_class.add(p)
+
+        for option in m2m_hatecrime:
+            o = HateCrimesandTrafficking.objects.get(hatecrimes_trafficking_option=option)
+            r.hatecrimes_trafficking.add(o)
+
+        r.assigned_section = r.assign_section()
+        r.save()
+        # adding this back for the save page results
+        form_data_dict['protected_class'] = m2m_protected_class.values()
+        form_data_dict['hatecrimes_trafficking'] = m2m_hatecrime.values()
+
+        return HttpResponseRedirect(f'/form/view/{r.pk}')
```

Comment:
This looks to be largely a duplicate of `CRTReportWizard.done`, good opportunity to pull this custom save logic into it's own function or maybe `Report.save` and call where needed.

Also a great spot and opportunity to add a [get_absolute_url](https://docs.djangoproject.com/en/3.0/ref/models/instances/#get-absolute-url) function to `Report` and use that here. We could definitely use that elsewhere.